omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java

```diff
@@ -0,0 +1,116 @@
+        + "/logentry", supportedClass = LogEntry.class, supportedOpenmrsVersions = { "1.9.*", "1.10.*", "1.11.*",
+                "1.12.*", "2.0.*", "2.1.*" })
+public class LogEntryResource extends DelegatingCrudResource<LogEntry> {
+
+    @Override
+    protected NeedsPaging<LogEntry> doGetAll(RequestContext context) {
+        return new NeedsPaging<LogEntry>(
+                Context.getService(IdentifierSourceService.class).getLogEntries(null, null, null, null, null, null),
+                context);
+    }
+
+    @Override
+    protected PageableResult doSearch(RequestContext context) {
+        IdentifierSourceService identifierSourceService = Context.getService(IdentifierSourceService.class);
+        UserService service = Context.getUserService();
+
+        String source = context.getRequest().getParameter("source");
+        String fromDate = context.getRequest().getParameter("fromDate");
+        String toDate = context.getRequest().getParameter("toDate");
+        String identifier = context.getRequest().getParameter("identifier");
+        String comment = context.getRequest().getParameter("comment");
+        String generatedBy = context.getRequest().getParameter("generatedBy");
+
+        List<LogEntry> logEntries = new ArrayList<LogEntry>();
+
+        IdentifierSource logSource = source != null ? identifierSourceService.getIdentifierSourceByUuid(source) : null;
+        Date dateFrom = fromDate != null ? (Date) ConversionUtil.convert(fromDate, Date.class) : null;
+        Date dateTo = toDate != null ? (Date) ConversionUtil.convert(toDate, Date.class) : null;
+        User user = generatedBy != null ? service.getUserByUuid(generatedBy) : null;
+
+        logEntries = identifierSourceService.getLogEntries(logSource, dateFrom, dateTo, identifier, user, comment);
+        return new NeedsPaging<LogEntry>(logEntries, context);
+    }
+
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+        DelegatingResourceDescription description = null;
+        if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+			description = new DelegatingResourceDescription();
+			
+			// metadata
+			description.addProperty("source");
+			description.addProperty("identifier");
+			description.addProperty("comment");
+			description.addProperty("generatedBy");
+            description.addProperty("dateGenerated");
+
+        }
+        return description;
+    }
+
+    @Override
+    public LogEntry newDelegate() throws ResourceDoesNotSupportOperationException {
+        throw new UnsupportedOperationException("method not supported");
+
+    }
+
+    @Override
+    public LogEntry save(LogEntry delegate) throws ResourceDoesNotSupportOperationException {
+        		throw new UnsupportedOperationException("save not supported");
+
+    }
+
+    @Override
+    public LogEntry getByUniqueId(String uniqueId) {
+        return null;
```

Comment:
Here can you assume that the value contained in uniqueId is the internal integer id and hence fetch by it?

Reply:
Do you mean using it as the log entry identifier?

Reply:
yes